code/client/munkilib/appleupdates.py

```diff
@@ -640,7 +640,7 @@ def _GetPreferredLocalization(self, list_of_localizations):
         localizations. Returns a single language/localization name.'''
         # pylint: disable=no-self-use
         localization_preferences = (
-            munkicommon.pref('AppleSoftwareUpdateLanguages') or ['English'])
+            munkicommon.getAppleLanguages() or ['English'])
```

Comment:
the behaivior is as you expected:
Johann is logged in when managedsoftwareuodate is running... munki get German dist files.
then Jean-Louis logs in and the dist are still in German... until munki runs in the background then munki get French dists